command.go

```diff
@@ -68,24 +68,39 @@ var ErrIncorrectType = errors.New("The command returned a value with a different
 
 // Call invokes the command for the given Request
 func (c *Command) Call(req *Request, re ResponseEmitter, env Environment) {
-	// we need the named return parameter so we can change the value from defer()
-	defer re.Close()
+	var err error
+
+	defer func() {
```

Comment:
How about putting all the logic in `Call` in a separate `call` function that turns an error. That way, we don't need this defer? The shared `err` variable makes be uncomfortable.